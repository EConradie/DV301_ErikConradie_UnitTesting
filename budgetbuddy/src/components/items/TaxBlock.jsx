import React from 'react'

function TaxBlock(props) {
    const {tax} = props
    
  return (
    <div data-testid="tax-card" className='card card-col p-2 row-tax'>
        <div className='card-row'>
            <span>
                <div aria-label='icon' className='avatar'>
                    {tax.icon}
                </div>
                <p aria-label='name'>{tax.name}</p>
            </span>

            <h4 aria-label='bracket'>{tax.bracket}%</h4>
        </div>
       
        <p aria-label='taxamount' className='amount'>- R {tax.taxAmount.toFixed(2)}</p>
    </div>
  )
}

export default TaxBlock