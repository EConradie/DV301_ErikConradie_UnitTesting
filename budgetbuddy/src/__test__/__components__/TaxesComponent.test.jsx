import { render, screen } from "@testing-library/react";
import TaxBlock from "../../components/items/TaxBlock";
import Taxes from "../../components/Taxes";

test("Test default state of TaxesBlock Component", () => {
  const taxesMock = {
    icon: "ðŸ“ˆ",
    name: "Income Tax",
    taxAmount: 300,
    bracket: 15,
  };

  render(<TaxBlock tax={taxesMock} />);

  const fullComponent = screen.getByTestId("tax-card");
  const icon = screen.getByLabelText("icon");
  const name = screen.getByLabelText("name");
  const bracket = screen.getByLabelText("bracket");
  const taxAmount = screen.getByLabelText("taxamount");

  expect(bracket).toHaveTextContent(
    `${taxesMock.bracket}%`
  );
  expect(taxAmount).toHaveTextContent(`- R ${taxesMock.taxAmount.toFixed(2)}`);
  expect(icon).toBeInTheDocument();
  expect(name).toHaveTextContent(taxesMock.name);
  expect(fullComponent).toMatchSnapshot();
});

test("Test if tax cards display when 2 tax entries are added", () => {
  const taxesMock = [
    {
      icon: "ðŸ“ˆ",
      name: "Income Tax",
      taxAmount: 300,
      bracket: 15,
    },
    {
      icon: "ðŸ“‰",
      name: "Property Tax",
      taxAmount: 500,
      bracket: 10,
    },
  ];

  render(
    <div>
      {taxesMock.map((tax, index) => (
        <TaxBlock key={index} tax={tax} />
      ))}
    </div>
  );

  const iconDisplay = screen.getAllByLabelText("icon");
  const nameDisplay = screen.getAllByLabelText("name");
  const bracketDisplay = screen.getAllByLabelText("bracket");
  const taxAmountDisplay = screen.getAllByLabelText("taxamount");

  expect(iconDisplay).toHaveLength(2);
  expect(nameDisplay).toHaveLength(2);
  expect(bracketDisplay).toHaveLength(2);
  expect(taxAmountDisplay).toHaveLength(2);

  nameDisplay.forEach((element, index) => {
    expect(element).toHaveTextContent(taxesMock[index].name);
  });

  iconDisplay.forEach((element, index) => {
    expect(element).toHaveTextContent(taxesMock[index].icon);
  });

  bracketDisplay.forEach((element, index) => {
    expect(element).toHaveTextContent(
      `${taxesMock[index].bracket}%`
    );
  });

  taxAmountDisplay.forEach((element, index) => {
    expect(element).toHaveTextContent(`- R ${taxesMock[index].taxAmount.toFixed(2)}`);
  });
});

test("Test if Taxes component renders correctly", () => {
  const incomeMock = [
    { icon: "ðŸ“ˆ", name: "Income Tax", taxAmount: 300, bracket: 15 },
    { icon: "ðŸ’°", name: "Sales Tax", taxAmount: 150, bracket: 10 },
  ];

  render(<Taxes income={incomeMock} />);

  const taxesComponent = screen.getByTestId("taxes-component");
  const taxBlocks = screen.getAllByTestId("tax-card");

  expect(taxesComponent).toBeInTheDocument();
  expect(taxBlocks).toHaveLength(incomeMock.length);
});
