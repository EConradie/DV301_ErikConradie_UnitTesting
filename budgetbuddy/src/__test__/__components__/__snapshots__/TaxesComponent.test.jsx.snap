// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test default state of TaxesBlock Component 1`] = `
<div
  class="card card-col p-2 row-tax"
  data-testid="tax-card"
>
  <div
    class="card-row"
  >
    <span>
      <div
        aria-label="icon"
        class="avatar"
      >
        ðŸ“ˆ
      </div>
      <p
        aria-label="name"
      >
        Income Tax
      </p>
    </span>
    <h4
      aria-label="bracket"
    >
      15
      %
    </h4>
  </div>
  <p
    aria-label="taxamount"
    class="amount"
  >
    - R 
    300.00
  </p>
</div>
`;
